# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/d3d/RendererD3D.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/renderer/d3d/RendererD3D.h
--- a/gfx/angle/checkout/src/libANGLE/renderer/d3d/RendererD3D.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/d3d/RendererD3D.h	Sat Sep 07 13:38:36 2019 +0000
@@ -22,7 +22,7 @@
 #include "libANGLE/renderer/d3d/VertexDataManager.h"
 #include "libANGLE/renderer/d3d/formatutilsD3D.h"
 #include "libANGLE/renderer/renderer_utils.h"
-#include "platform/WorkaroundsD3D.h"
+#include "platform/FeaturesD3D.h"
 
 namespace egl
 {
@@ -182,6 +182,7 @@
                                          IUnknown *d3dTexture,
                                          EGLint *width,
                                          EGLint *height,
+                                         EGLint *samples,
                                          const angle::Format **angleFormat) const  = 0;
     virtual egl::Error validateShareHandle(const egl::Config *config,
                                            HANDLE shareHandle,
@@ -189,7 +190,7 @@
 
     virtual int getMajorShaderModel() const = 0;
 
-    const angle::WorkaroundsD3D &getWorkarounds() const;
+    const angle::FeaturesD3D &getFeatures() const;
 
     // Pixel operations
     virtual angle::Result copyImage2D(const gl::Context *context,
@@ -274,7 +275,12 @@
     virtual UniformStorageD3D *createUniformStorage(size_t storageSize) = 0;
 
     // Image operations
-    virtual ImageD3D *createImage()                                                        = 0;
+    virtual ImageD3D *createImage() = 0;
+    virtual ExternalImageSiblingImpl *createExternalImageSibling(
+        const gl::Context *context,
+        EGLenum target,
+        EGLClientBuffer buffer,
+        const egl::AttributeMap &attribs)                                                  = 0;
     virtual angle::Result generateMipmap(const gl::Context *context,
                                          ImageD3D *dest,
                                          ImageD3D *source)                                 = 0;
@@ -343,7 +349,7 @@
                                                   const gl::Box &destArea)    = 0;
 
     // Device lost
-    GLenum getResetStatus();
+    gl::GraphicsResetStatus getResetStatus();
     void notifyDeviceLost();
     virtual bool resetDevice()          = 0;
     virtual bool testDeviceLost()       = 0;
@@ -380,7 +386,8 @@
     // Necessary hack for default framebuffers in D3D.
     virtual FramebufferImpl *createDefaultFramebuffer(const gl::FramebufferState &state) = 0;
 
-    virtual gl::Version getMaxSupportedESVersion() const = 0;
+    virtual gl::Version getMaxSupportedESVersion() const  = 0;
+    virtual gl::Version getMaxConformantESVersion() const = 0;
 
     angle::Result initRenderTarget(const gl::Context *context, RenderTargetD3D *renderTarget);
 
@@ -408,7 +415,7 @@
   private:
     void ensureCapsInitialized() const;
 
-    virtual angle::WorkaroundsD3D generateWorkarounds() const = 0;
+    virtual void initializeFeatures(angle::FeaturesD3D *features) const = 0;
 
     mutable bool mCapsInitialized;
     mutable gl::Caps mNativeCaps;
@@ -416,8 +423,8 @@
     mutable gl::Extensions mNativeExtensions;
     mutable gl::Limitations mNativeLimitations;
 
-    mutable bool mWorkaroundsInitialized;
-    mutable angle::WorkaroundsD3D mWorkarounds;
+    mutable bool mFeaturesInitialized;
+    mutable angle::FeaturesD3D mFeatures;
 
     bool mDisjoint;
     bool mDeviceLost;